nsqd/protocol_v2.go

```diff
@@ -318,6 +316,7 @@ func (p *protocolV2) messagePump(client *clientV2, startedChan chan bool) {
 
 			subChannel.StartInFlightTimeout(msg, client.ID, msgTimeout)
 			client.SendingMessage()
+			var buf bytes.Buffer
 			err = p.SendMessage(client, msg, &buf)
```

Comment:
Since we're no longer re-using the buf, you can simplify this further: declare it inside `SendMessage()`, and remove the argument.

Reply:
Done. PTAL. @ploxiln